src/ui/components/RecommendedBadge/index.js

```diff
@@ -22,6 +22,10 @@ type InternalProps = {|
 export const RecommendedBadgeBase = ({ i18n, size }: InternalProps) => {
   const label = i18n.gettext('Recommended');
 
+  const onClick = (event: SyntheticEvent<any>) => {
```

Comment:
This is risky because you might not want to stop propagation in all situations. I suggest making it a prop so you can just add the `onClick` handler for `SearchResult` only.

Example:

```diff
diff --git a/src/amo/components/SearchResult/index.js b/src/amo/components/SearchResult/index.js
index 1f1ad5526..049cfc6b2 100644
--- a/src/amo/components/SearchResult/index.js
+++ b/src/amo/components/SearchResult/index.js
@@ -172,7 +172,10 @@ export class SearchResultBase extends React.Component<InternalProps> {
             addon &&
             addon.is_recommended &&
             clientApp !== CLIENT_APP_ANDROID ? (
-              <RecommendedBadge size="small" />
+              <RecommendedBadge
+                onClick={(e) => e.stopPropagation()}
+                size="small"
+              />
             ) : null}
           </h2>
           {summary}
```